sklearn/metrics/pairwise.py

```diff
@@ -527,6 +540,53 @@ def manhattan_distances(X, Y=None, sum_over_features=True,
     return D.reshape((-1, X.shape[1]))
 
 
+def _center_and_normalize(X, copy=False):
```

Comment:
I am very surprised that we don't already have a function that does that in sklearn.